Improve test failure readability

ilmotta opened the pull request:
### Summary

This PR is one of those you regret not opening before, way before. **Problem**: failed equality checks as in `(is (= expected actual))` will give a single, long line of output that for anything but the simplest data structures is unreadable by humans, and the output doesn't give a useful diff. For me, unit testing so far in status-mobile has been one of the most insufferable experiences ever because of that alone.

The solution is simple: use library https://github.com/nubank/matcher-combinators which will define a directive `match?` which will pinpoint where two data structures differ. Then, instead of `(is (= ...`, use `(is (match? expected actual)`. It works beautifully. ~~No need to require any extra namespace, the directives are injected once in `status-im.test-runner`~~ (false, in the CI the ClojureScript compiler couldn't find the `match?` directive).

Without this PR, if a test fails, I often had to copy the output, format it with zprint and diff two buffers and still manually scan the output hunting for a diff. It could be automated more, but it's just not worth it and it doesn't solve the problem for other devs.

### Note

Guidelines were updated too: https://github.com/status-im/status-mobile/blob/ilmotta/improve-readability-of-test-failures/doc/new-guidelines.md#prefer-match-over--when-comparing-data-structures.

The library offers other nice matchers, but the majority of the time `match?` is sufficient.

### Can we use another test runner like Kaocha?

[kaocha-cljs2](https://github.com/lambdaisland/kaocha-cljs2) would be able to print better test errors out of the box, among other features, but I have no clue if it would work well or at all in our stack (in theory yes, but it's a larger piece of work).

### Demo

<img src="https://github.com/status-im/status-mobile/assets/46027/1cb849f8-3fc0-4e4e-a742-e057651cd38f" width="700" />

<img src="https://github.com/status-im/status-mobile/assets/46027/54291cb2-d6f3-4f14-b50f-1f1f25f13a39" width="700" />

#### Areas that maybe impacted

None, just a few test namespaces were changed from using `(is (= ...)` to `(is (match? ...)`

status: ready

ilmotta committed:
Add lib nubank/matcher-combinators and use in some unit tests

ilmotta committed:
Update clojure deps

ilmotta committed:
Update guidelines

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :x: | 3366ffe4 | [#1](https://ci.status.im/job/status-mobile/job/prs/job/tests/job/PR-18049/1/) | 2023-12-01 16:13:48 | ~3 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-mobile/job/prs/job/tests/job/PR-18049/1/consoleText) |

clauxx reviewed (approved):
Thank you, looks great! Had to do the diffing manually once and it wasn't pleasant

ilmotta committed:
Move require for matcher-combinators.test where warnings are not generated

ilmotta committed:
Fix: ClojureScript compiler can't find match? directive

The namespace matcher-combinators.test must be required in namespaces using
"match?"

jakubgs reviewed (approved):
Approving only Nix related changes.

status-im-auto commented:
## 77% of end-end tests have passed
```
Total executed tests: 48
Failed tests: 4
Expected to fail tests: 7
Passed tests: 37
```
```
IDs of failed tests: 702733,703133,702948,702869 
```
```
IDs of expected to fail tests: 702732,702894,702783,703503,702846,702731,702808 
```
<h3>Failed tests (4)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=develop&PR_ID=18049&TR_CASE_IDS=702733,703133,702948,702869">Rerun failed tests</a></li><br/><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3297911">test_1_1_chat_text_message_delete_push_disappear</a>, id: 702733 </b></td></tr><tr><td><p><blockquote><div>Device 1: Getting PN by 'DELETE ME' </div><div>Device 2: Could not reach home view by pressing system back button </div></blockquote></p>

```
critical/chats/test_1_1_public_chats.py:516: in test_1_1_chat_text_message_delete_push_disappear
    self.errors.verify_no_errors()
base_test_case.py:191: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Deleted message is shown in chat view for 1-1 chat
E    Deleted message is shown in chat view for 1-1 chat
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/1eff7b8e48854bcbb9c184666b5e22bf?auth=6d52b2b318b2392ed7f6c14870a30af9#408">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/406ee913eab149479c968210d6a41f38?auth=07baa1c601676cdabea4a5dccf2cfc5d#349">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityMultipleDeviceMergedTwo:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3297940">test_community_hashtag_links_to_community_channels</a>, id: 702948 </b></td></tr><tr><td><p><blockquote><div>Device 2: Find `Text` by `xpath`: `//android.view.ViewGroup[@content-desc='chat-item']//android.widget.TextView[contains(@text,'#cats')]` </div><div>Device 2: Looking for a message by text: it is just a message text </div></blockquote></p>

```
critical/chats/test_public_chat_browsing.py:1046: in test_community_hashtag_links_to_community_channels
    self.errors.verify_no_errors()
base_test_case.py:191: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Receiver was not navigated to the cats channel
E    Sender was not navigated to the cats channel
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/992feb5cbd6e4b738f130594ed94bb4d?auth=65589dd615fd2c346582af13a3adcf72#324">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/87988832a90d44f4968867eb87c64f4c?auth=a7bc7af20afbca2fda86c0ad5806a034#321">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3297944">test_restore_multiaccount_with_waku_backup_remove_switch</a>, id: 703133 </b></td></tr><tr><td><p><blockquote><div>Device 1: Tap on found: Button </div><div>Device 1: Wait for element `Button` for max 30s and click when it is available </div></blockquote></p>

```
critical/chats/test_public_chat_browsing.py:194: in test_restore_multiaccount_with_waku_backup_remove_switch
    self.sign_in.recover_access(passphrase=waku_user.seed, second_user=True)
../views/sign_in_view.py:276: in recover_access
    self.identifiers_button.wait_and_click(30)
../views/base_element.py:96: in wait_and_click
    self.wait_for_visibility_of_element(sec)
../views/base_element.py:139: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 1: Button by accessibility id:`skip-identifiers` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/3a6f93c0385c4e7f964668952117b7cd?auth=c9be450b6c68b7f7d45123d24ac548ea#230">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3297936">test_community_undo_delete_message</a>, id: 702869 </b></td></tr><tr><td><p><blockquote><div>Device 1: Tap on found: Button </div><div>Device 1: Looking for a message by text: message to delete and undo </div></blockquote></p>

```
critical/chats/test_public_chat_browsing.py:113: in test_community_undo_delete_message
    self.channel.chat_element_by_text(message_to_delete).wait_for_visibility_of_element()
../views/base_element.py:139: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 1: ChatElementByText by xpath:`//*[starts-with(@text,'message to delete and undo')]/ancestor::android.view.ViewGroup[@content-desc='chat-item']` is not found on the screen after wait_for_visibility_of_element

During handling of the above exception, another exception occurred:
critical/chats/test_public_chat_browsing.py:115: in test_community_undo_delete_message
    pytest.fail("Message was not restored by clicking 'Undo' button")
E   Failed: Message was not restored by clicking 'Undo' button
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/3a6f93c0385c4e7f964668952117b7cd?auth=c9be450b6c68b7f7d45123d24ac548ea#84">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p></ul></p></td></tr></tbody></table></details><h3>Expected to fail tests (7)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3297909">test_1_1_chat_pin_messages</a>, id: 702731 </b></td></tr><tr><td>

```
Test is not run, e2e blocker  
```

<b>[[reason: [NOTRUN] Pin feature is in development]]</b><br/><br/></td></tr></tbody></table><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3297937">test_community_contact_block_unblock_offline</a>, id: 702894 </b></td></tr><tr><td><p><blockquote><div>Device 1: Find `ProfileSendMessageButton` by `accessibility id`: `Chat-item-button` </div><div>Device 1: Sending message 'piy' </div></blockquote></p>

```
critical/chats/test_public_chat_browsing.py:741: in test_community_contact_block_unblock_offline
    self.errors.verify_no_errors()
base_test_case.py:191: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Chat with unblocked user was not enabled after 1 minute 
```

<b>[[Message can be missed after unblock: https://github.com/status-im/status-mobile/issues/16873]]</b><br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/7a726d07da5343a1af73175a244f5d2d?auth=b005f7fe8d803a839d52da9cc5fdb489#385">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/2451dac92e4343e58e114b5defa31703?auth=b1e980bd4bbc9dd65c1c332cb7cfb764#352">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevicesNewUiTwo:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3297917">test_1_1_chat_is_shown_message_sent_delivered_from_offline</a>, id: 702783 </b></td></tr><tr><td><p><blockquote><div>Device 2: Find `Text` by `xpath`: `//*[starts-with(@text,'test message')]/ancestor::android.view.ViewGroup[@content-desc='chat-item']//*[@content-desc='message-status']/android.widget.TextView` </div><div>Device 2: `Text` is `Sent` </div></blockquote></p>

```
critical/chats/test_1_1_public_chats.py:620: in test_1_1_chat_is_shown_message_sent_delivered_from_offline
    self.errors.verify_no_errors()
base_test_case.py:191: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Message status was not changed to Delivered, it's Sent after back up online! 
```

<b>[[Data delivery issue]]</b><br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/60b38d731ae24cdc99368e38a0dc613d?auth=0c5ce5dd6e9e0b2a3debb647f696dc53#81">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/00161678ca3a46409496231b0fdce250?auth=4cfcd35d8856baa791e11b4575846386#81">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3297952">test_community_discovery</a>, id: 703503 </b></td></tr><tr><td>

```
Test is not run, e2e blocker  
```

<b>[[reason: [NOTRUN] Curated communities not loading, https://github.com/status-im/status-mobile/issues/17852]]</b><br/><br/></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3297931">test_community_navigate_to_channel_when_relaunch</a>, id: 702846 </b></td></tr><tr><td>

```
Test is not run, e2e blocker  
```

<b>[[reason: [NOTRUN] Issue #18075]]</b><br/><br/></td></tr></tbody></table><h4>Class TestGroupChatMultipleDeviceMergedNewUI:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3297910">test_group_chat_pin_messages</a>, id: 702732 </b></td></tr><tr><td>

```
Test is not run, e2e blocker  
```

<b>[[reason: [NOTRUN] Pin feature is in development]]</b><br/><br/></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3297921">test_group_chat_offline_pn</a>, id: 702808 </b></td></tr><tr><td><p><blockquote><div>Device 3: Looking for a message by text: message from old member </div><div>Device 3: Looking for a message by text: message from new member </div></blockquote></p>

```
critical/chats/test_group_chat.py:323: in test_group_chat_offline_pn
    self.errors.verify_no_errors()
base_test_case.py:191: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Messages PN was not fetched from offline 
```

<b>[[Data delivery issue]]</b><br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/881921fb3f6440ff8f6ffb2619f67fc0?auth=9a1e43665c496eff50ce5ba1dc1b8094#305">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d578c783e57243a5870675023bf0c9d2?auth=253d28fc0981fc897d1fda954d587baa#211">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/b7dff0c89ea54636a9a3dcb59a12835e?auth=ef3627aadccf3ac41e2d33c6c6cb9a00#197">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p></ul></p></td></tr></tbody></table></details><h3>Passed tests (37)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestActivityMultipleDevicePRTwo:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3297941">test_activity_center_mentions</a>, id: 702957 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/f530ebd34bb54fdb838356d0a1276974?auth=1f254cde15ca26632e176b3b307b7638#119">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/8c14dea529524adf96ba02794370f620?auth=42833127b4bac172952160f041329502#98">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3297942">test_activity_center_admin_notification_accept_swipe</a>, id: 702958 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/f530ebd34bb54fdb838356d0a1276974?auth=1f254cde15ca26632e176b3b307b7638#167">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/8c14dea529524adf96ba02794370f620?auth=42833127b4bac172952160f041329502#112">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestDeepLinksOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3297954">test_links_open_universal_links_from_chat</a>, id: 704613 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a632ba2a22bc46e9a296c2dc447bcc41?auth=978cd01824caed3ac917567d48e00065#55">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3297955">test_links_open_universal_links_from_other_apps</a>, id: 704614 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a632ba2a22bc46e9a296c2dc447bcc41?auth=978cd01824caed3ac917567d48e00065#267">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3297914">test_links_deep_links</a>, id: 702775 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a632ba2a22bc46e9a296c2dc447bcc41?auth=978cd01824caed3ac917567d48e00065#136">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3297916">test_1_1_chat_emoji_send_reply_and_open_link</a>, id: 702782 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/1eff7b8e48854bcbb9c184666b5e22bf?auth=6d52b2b318b2392ed7f6c14870a30af9#131">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/406ee913eab149479c968210d6a41f38?auth=07baa1c601676cdabea4a5dccf2cfc5d#112">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3297923">test_1_1_chat_push_emoji</a>, id: 702813 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/1eff7b8e48854bcbb9c184666b5e22bf?auth=6d52b2b318b2392ed7f6c14870a30af9#263">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/406ee913eab149479c968210d6a41f38?auth=07baa1c601676cdabea4a5dccf2cfc5d#206">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3297913">test_1_1_chat_non_latin_messages_stack_update_profile_photo</a>, id: 702745 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/1eff7b8e48854bcbb9c184666b5e22bf?auth=6d52b2b318b2392ed7f6c14870a30af9#193">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/406ee913eab149479c968210d6a41f38?auth=07baa1c601676cdabea4a5dccf2cfc5d#155">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3297934">test_1_1_chat_edit_message</a>, id: 702855 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/1eff7b8e48854bcbb9c184666b5e22bf?auth=6d52b2b318b2392ed7f6c14870a30af9#320">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/406ee913eab149479c968210d6a41f38?auth=07baa1c601676cdabea4a5dccf2cfc5d#245">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3297949">test_1_1_chat_send_image_save_and_share</a>, id: 703391 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/1eff7b8e48854bcbb9c184666b5e22bf?auth=6d52b2b318b2392ed7f6c14870a30af9#348">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/406ee913eab149479c968210d6a41f38?auth=07baa1c601676cdabea4a5dccf2cfc5d#288">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>6. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3297908">test_1_1_chat_message_reaction</a>, id: 702730 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/1eff7b8e48854bcbb9c184666b5e22bf?auth=6d52b2b318b2392ed7f6c14870a30af9#81">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/406ee913eab149479c968210d6a41f38?auth=07baa1c601676cdabea4a5dccf2cfc5d#81">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3297945">test_community_several_images_send_reply</a>, id: 703194 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/7a726d07da5343a1af73175a244f5d2d?auth=b005f7fe8d803a839d52da9cc5fdb489#209">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/2451dac92e4343e58e114b5defa31703?auth=b1e980bd4bbc9dd65c1c332cb7cfb764#163">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3297935">test_community_one_image_send_reply</a>, id: 702859 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/7a726d07da5343a1af73175a244f5d2d?auth=b005f7fe8d803a839d52da9cc5fdb489#234">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/2451dac92e4343e58e114b5defa31703?auth=b1e980bd4bbc9dd65c1c332cb7cfb764#230">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3297926">test_community_emoji_send_copy_paste_reply</a>, id: 702840 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/7a726d07da5343a1af73175a244f5d2d?auth=b005f7fe8d803a839d52da9cc5fdb489#281">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/2451dac92e4343e58e114b5defa31703?auth=b1e980bd4bbc9dd65c1c332cb7cfb764#286">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3297943">test_community_mark_all_messages_as_read</a>, id: 703086 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/7a726d07da5343a1af73175a244f5d2d?auth=b005f7fe8d803a839d52da9cc5fdb489#550">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/2451dac92e4343e58e114b5defa31703?auth=b1e980bd4bbc9dd65c1c332cb7cfb764#405">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3297925">test_community_message_delete</a>, id: 702839 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/7a726d07da5343a1af73175a244f5d2d?auth=b005f7fe8d803a839d52da9cc5fdb489#192">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/2451dac92e4343e58e114b5defa31703?auth=b1e980bd4bbc9dd65c1c332cb7cfb764#130">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>6. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3297924">test_community_message_send_check_timestamps_sender_username</a>, id: 702838 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/7a726d07da5343a1af73175a244f5d2d?auth=b005f7fe8d803a839d52da9cc5fdb489#150">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/2451dac92e4343e58e114b5defa31703?auth=b1e980bd4bbc9dd65c1c332cb7cfb764#101">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>7. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3297929">test_community_links_with_previews_github_youtube_twitter_gif_send_enable</a>, id: 702844 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/7a726d07da5343a1af73175a244f5d2d?auth=b005f7fe8d803a839d52da9cc5fdb489#303">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/2451dac92e4343e58e114b5defa31703?auth=b1e980bd4bbc9dd65c1c332cb7cfb764#301">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>8. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3297928">test_community_message_edit</a>, id: 702843 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/7a726d07da5343a1af73175a244f5d2d?auth=b005f7fe8d803a839d52da9cc5fdb489#168">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/2451dac92e4343e58e114b5defa31703?auth=b1e980bd4bbc9dd65c1c332cb7cfb764#118">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>9. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3297927">test_community_unread_messages_badge</a>, id: 702841 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/7a726d07da5343a1af73175a244f5d2d?auth=b005f7fe8d803a839d52da9cc5fdb489#348">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/2451dac92e4343e58e114b5defa31703?auth=b1e980bd4bbc9dd65c1c332cb7cfb764#342">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestActivityCenterContactRequestMultipleDevicePR:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3297915">test_add_contact_field_validation</a>, id: 702777 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/318990beca2c4d13b7dc238a5d639c4b?auth=b0a4826ff8ad5f6cca0a0921712dc62c#133">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d6d52da67f26485aa5eb6e10325388f3?auth=17e292c2dde5652508b1df8176d1d216#193">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3297933">test_activity_center_contact_request_accept_swipe_mark_all_as_read</a>, id: 702851 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/318990beca2c4d13b7dc238a5d639c4b?auth=b0a4826ff8ad5f6cca0a0921712dc62c#93">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d6d52da67f26485aa5eb6e10325388f3?auth=17e292c2dde5652508b1df8176d1d216#96">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3297932">test_activity_center_contact_request_decline</a>, id: 702850 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/318990beca2c4d13b7dc238a5d639c4b?auth=b0a4826ff8ad5f6cca0a0921712dc62c#57">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d6d52da67f26485aa5eb6e10325388f3?auth=17e292c2dde5652508b1df8176d1d216#54">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityMultipleDeviceMergedTwo:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3297922">test_community_markdown_support</a>, id: 702809 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/992feb5cbd6e4b738f130594ed94bb4d?auth=65589dd615fd2c346582af13a3adcf72#183">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/87988832a90d44f4968867eb87c64f4c?auth=a7bc7af20afbca2fda86c0ad5806a034#165">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3297919">test_community_mentions_push_notification</a>, id: 702786 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/992feb5cbd6e4b738f130594ed94bb4d?auth=65589dd615fd2c346582af13a3adcf72#129">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/87988832a90d44f4968867eb87c64f4c?auth=a7bc7af20afbca2fda86c0ad5806a034#102">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3297930">test_community_leave</a>, id: 702845 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/992feb5cbd6e4b738f130594ed94bb4d?auth=65589dd615fd2c346582af13a3adcf72#321">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/87988832a90d44f4968867eb87c64f4c?auth=a7bc7af20afbca2fda86c0ad5806a034#290">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3297953">test_community_join_when_node_owner_offline</a>, id: 703629 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/992feb5cbd6e4b738f130594ed94bb4d?auth=65589dd615fd2c346582af13a3adcf72#540">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/87988832a90d44f4968867eb87c64f4c?auth=a7bc7af20afbca2fda86c0ad5806a034#487">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestGroupChatMultipleDeviceMergedNewUI:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3297950">test_group_chat_mute_chat</a>, id: 703495 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/881921fb3f6440ff8f6ffb2619f67fc0?auth=9a1e43665c496eff50ce5ba1dc1b8094#357">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d578c783e57243a5870675023bf0c9d2?auth=253d28fc0981fc897d1fda954d587baa#237">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/b7dff0c89ea54636a9a3dcb59a12835e?auth=ef3627aadccf3ac41e2d33c6c6cb9a00#231">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3297947">test_group_chat_send_image_save_and_share</a>, id: 703297 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/881921fb3f6440ff8f6ffb2619f67fc0?auth=9a1e43665c496eff50ce5ba1dc1b8094#261">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d578c783e57243a5870675023bf0c9d2?auth=253d28fc0981fc897d1fda954d587baa#172">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/b7dff0c89ea54636a9a3dcb59a12835e?auth=ef3627aadccf3ac41e2d33c6c6cb9a00#136">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3297946">test_group_chat_reactions</a>, id: 703202 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/881921fb3f6440ff8f6ffb2619f67fc0?auth=9a1e43665c496eff50ce5ba1dc1b8094#154">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d578c783e57243a5870675023bf0c9d2?auth=253d28fc0981fc897d1fda954d587baa#103">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/b7dff0c89ea54636a9a3dcb59a12835e?auth=ef3627aadccf3ac41e2d33c6c6cb9a00#83">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3297920">test_group_chat_join_send_text_messages_push</a>, id: 702807 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/881921fb3f6440ff8f6ffb2619f67fc0?auth=9a1e43665c496eff50ce5ba1dc1b8094#135">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d578c783e57243a5870675023bf0c9d2?auth=253d28fc0981fc897d1fda954d587baa#75">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/b7dff0c89ea54636a9a3dcb59a12835e?auth=ef3627aadccf3ac41e2d33c6c6cb9a00#75">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3297912">test_community_copy_and_paste_message_in_chat_input</a>, id: 702742 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/3a6f93c0385c4e7f964668952117b7cd?auth=c9be450b6c68b7f7d45123d24ac548ea#55">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3297948">test_community_mute_community_and_channel</a>, id: 703382 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/3a6f93c0385c4e7f964668952117b7cd?auth=c9be450b6c68b7f7d45123d24ac548ea#104">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestActivityMultipleDevicePR:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3297938">test_navigation_jump_to</a>, id: 702936 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/df9bba33e23942f6bc928cf556d9b3ca?auth=5afa165262dc8abc2269b5d8e3173587#139">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9e01fd69f5e2419299416c94b6e28092?auth=3b09730ee5d16adad6905e8cad6e8f55#117">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3297939">test_activity_center_reply_read_unread_delete_filter_swipe</a>, id: 702947 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/df9bba33e23942f6bc928cf556d9b3ca?auth=5afa165262dc8abc2269b5d8e3173587#162">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9e01fd69f5e2419299416c94b6e28092?auth=3b09730ee5d16adad6905e8cad6e8f55#120">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevicesNewUiTwo:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3297918">test_1_1_chat_delete_via_long_press_relogin</a>, id: 702784 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/60b38d731ae24cdc99368e38a0dc613d?auth=0c5ce5dd6e9e0b2a3debb647f696dc53#183">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/00161678ca3a46409496231b0fdce250?auth=4cfcd35d8856baa791e11b4575846386#268">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3297951">test_1_1_chat_mute_chat</a>, id: 703496 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/60b38d731ae24cdc99368e38a0dc613d?auth=0c5ce5dd6e9e0b2a3debb647f696dc53#111">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/00161678ca3a46409496231b0fdce250?auth=4cfcd35d8856baa791e11b4575846386#255">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details>
